apps/emqx_exhook/src/emqx_exhook.erl

```diff
@@ -78,37 +78,50 @@ cast(Hookpoint, Req, [ServerName|More]) ->
 
 -spec call_fold(atom(), term(), function())
   -> {ok, term()}
-   | {stop, term()}.
+   | {stop, term()}
+   | ignore.
 call_fold(Hookpoint, Req, AccFun) ->
     FailedAction = emqx_exhook_mngr:get_request_failed_action(),
     ServerNames = emqx_exhook_mngr:running(),
     case ServerNames == [] andalso FailedAction == deny of
         true ->
+            ?LOG(warning, "No available Server. Stop hook chain execute."),
             {stop, deny_action_result(Hookpoint, Req)};
         _ ->
-            call_fold(Hookpoint, Req, FailedAction, AccFun, ServerNames)
+            InitAcc = Req, %% `Req` (includede message.. etc.) as `InitAcc`
```

Comment:
Keep the original style?

Reply:
It's actually an `Accumulator` for `emqx_hook` system.
I'll Keep original style with an extra comment.